packages/strapi-connector-bookshelf/lib/queries.js

```diff
@@ -95,19 +103,15 @@ module.exports = function createQueryBuilder({ model, strapi }) {
     }
 
     const runCreate = async trx => {
-      try {
-        // Create entry with no-relational data.
-        const entry = await model.forge(data).save(null, { transacting: trx });
-        const isDraft = contentTypesUtils.isDraft(entry.toJSON(), model);
-        await createComponents(entry, attributes, { transacting: trx, isDraft });
+      // Create entry with no-relational data.
+      const entry = await model.forge(data).save(null, { transacting: trx });
+      const isDraft = contentTypesUtils.isDraft(entry.toJSON(), model);
+      await createComponents(entry, attributes, { transacting: trx, isDraft });
 
-        return model.updateRelations({ id: entry.id, values: relations }, { transacting: trx });
-      } catch (err) {
-        handleDatabaseError(err);
-      }
+      return model.updateRelations({ id: entry.id, values: relations }, { transacting: trx });
     };
 
-    return wrapTransaction(runCreate, { transacting });
+    return wrapErrors(wrapTransaction(runCreate, { transacting }));
```

Comment:
I think here it needs to be `wrapErrors(() => wrapTransaction(runCreate, { transaction}))`

Or just do like you did in the mongoose queries :)

Reply:
I had to wrap `runCreate` directly instead of `wrapTransaction` as this function is not asynchronous and it was causing some issues 🤷‍♂️